tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -656,6 +656,226 @@ def test_log_null_metric(self):
+        assert metric_obj.step == 3
+        assert metric_obj.timestamp == 50
+        assert metric_obj.value == 20
+
+    def test_log_metrics_multiple_names(self):
+
+        run = self._run_factory()
+        run_id = run.info.run_id
+        metric_name_a = "test-metric-a"
+        metric_name_b = "test-metric-b"
+        metric_name_c = "test-metric-c"
+
+        metric_tuples = [
+            # Parameters are (metric name, value, step, timestamp)
+            (metric_name_a, 10, 0, 0),
+            (metric_name_b, 1, 0, 0),
+            (metric_name_a, 20, 1, 1),
+            (metric_name_b, 2, 1, 1),
+            (metric_name_a, 50, 5, 5),  # latest step for metric a
+            (metric_name_b, 2, 5, 5),
+            (metric_name_b, 3, 6, 6),
+            (metric_name_b, 4, 7, 7),
+            (metric_name_b, 6, 2, 2),
+            (metric_name_b, 7, 8, 8),  # latest step for metric b
+            (metric_name_b, 8, 1, 1),
+            (metric_name_a, 100, 4, 4),
+            (metric_name_a, 120, 4, 4),
+            (metric_name_c, -10, 3, 3),
+        ]
+
+        metric_a_tuples = [m for m in metric_tuples if m[0] == metric_name_a]
+        metric_b_tuples = [m for m in metric_tuples if m[0] == metric_name_b]
+        metric_c_tuples = [m for m in metric_tuples if m[0] == metric_name_c]
+
+        metric_entities = [Metric(*metric_tuple) for metric_tuple in metric_tuples]
+
+        self.store.log_metrics(run_id, metric_entities)
+        metric_a_histories = self.store.get_metric_history(run_id, metric_name_a)
+        metric_b_histories = self.store.get_metric_history(run_id, metric_name_b)
+        metric_c_histories = self.store.get_metric_history(run_id, metric_name_c)
+        assert len(metric_a_histories) == 5
+        assert len(metric_b_histories) == 8
+        assert len(metric_c_histories) == 1
+
+        metrics_a = [(m.key, m.value, m.timestamp, m.step) for m in metric_a_histories]
+        metrics_b = [(m.key, m.value, m.timestamp, m.step) for m in metric_b_histories]
+        metrics_c = [(m.key, m.value, m.timestamp, m.step) for m in metric_c_histories]
+        assert set(metrics_a) == set(metric_a_tuples)
+        assert set(metrics_b) == set(metric_b_tuples)
+        assert set(metrics_c) == set(metric_c_tuples)
+
+        run_data = self.store.get_run(run_id).data
+        run_metrics = run_data.metrics
+
+        assert len(run_metrics) == 3
+        assert run_metrics[metric_name_a] == 50
+        assert run_metrics[metric_name_b] == 7
+        assert run_metrics[metric_name_c] == -10
+
+        last_metric_tuples = [
+            (metric_name_a, 50, 5, 5),
+            (metric_name_b, 7, 8, 8),
+            (metric_name_c, -10, 3, 3),
+        ]
+        last_metrics = [(m.key, m.value, m.timestamp, m.step) for m in run_data._metric_objs]
+        assert set(last_metrics) == set(last_metric_tuples)
+
+    def test_log_metrics_multiple_names_deactivated_batch_mode(self):
```

Comment:
same comment as before. i am not sure how this is deactivating batch_mode

Reply:
There is a comment in line 848. Please also see the previous comment.